hydroflow/src/scheduled/graph.rs

```diff
@@ -409,8 +412,79 @@ impl Hydroflow {
             _phantom: PhantomData,
         }
     }
+
+    pub fn add_dependencies(&mut self, subg_id: SubgraphId, deps: DirectedEdgeSet) {
+        let base = self.subgraphs[subg_id].dependencies.current_id;
+        for (key, val) in deps.node_names {
+            self.subgraphs[subg_id]
+                .dependencies
+                .node_names
+                .insert(key + base, val);
+        }
+        for (from, to) in deps.edges {
+            self.subgraphs[subg_id]
+                .dependencies
+                .edges
+                .insert((from + base, to + base));
+        }
+        self.subgraphs[subg_id].dependencies.current_id += deps.current_id;
```

Comment:
Better to put this in a method on `DirectedEdgeSet`, similar to https://doc.rust-lang.org/std/vec/struct.Vec.html#method.append . Then this will just be `self.subgraphs[sg_id].dependencies.append(deps)` or similar